text/0000-cargo-libstd-awareness.md

```diff
@@ -0,0 +1,154 @@
+- Feature Name: cargo_libstd_awareness
+- Start Date: 2015-05-26
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+
+Currently, Cargo doesn't know whether packages depend on libstd. This makes Cargo unsuitable for
+packages that need a cross-compiled or custom libstd, or otherwise depend on crates with the same
+names as libstd and the crates behind the facade. The proposed fixes also open the door to a future
+where libstd can be Cargoized.
+
+
+# Motivation
+
+First some background. The current situation seems to be more of an accident of `rustc`'s pre-Cargo
+history than an explicit design decision. Cargo passes the location and name of all depended-on
+crates to `rustc`. This method is good for a number of reasons stemming from its fine granularity,
+such as:
+
+ - No undeclared dependencies can be used
+
+ - Conversely, `rustc` can warn against *unused* declared dependencies
+
+ - Crate/symbol names are frobbed so that packages with the overlapping names don't conflict
+
+
+However rather than passing in libstd and its deps, Cargo lets the compiler look for them as need in
+the compiler's sysroot [specifically `<sysroot>/lib/<target>`]. This is quite coarse in comparison,
```

Comment:
I'm a bit confused what you'd like me to elaborate on. I already wanted to emphasize we just do this in the case of libstd and its dependencies---in other words that we are so close---just 1 library away!---from not linking with the sysroot at all.